flask_monitoringdashboard/database/function_calls.py

```diff
@@ -4,73 +4,50 @@
+    """
+    :param endpoint: name of the endpoint
+    :param list_of_days: a list with datetime.date objects
+    :return: A list of the amount of requests per day for a specific endpoint.
+    If no requests is made on that day, a value of 0 is returned.
+    """
+    result_list = []
     with session_scope() as db_session:
-        query = text("""select strftime('%Y-%m-%d', time) AS newTime,
-                               count(endpoint) AS cnt,
-                               endpoint
-                        from functioncalls 
-                        group by newTime, endpoint""")
-        result = db_session.execute(query)
-        data = result.fetchall()
-        return data
+        for day in list_of_days:
+            result = db_session.query(FunctionCall.execution_time). \
+                filter(FunctionCall.endpoint == endpoint). \
+                filter(func.strftime('%Y-%m-%d', FunctionCall.time) == day.strftime('%Y-%m-%d')).count()
+            result_list.append(result)
+    return result_list
 
 
-def add_function_call(time, endpoint, ip):
+def add_function_call(execution_time, endpoint, ip):
     """ Add a measurement to the database. """
     with session_scope() as db_session:
-        group_by = None
-        try:
-            if config.get_group_by:
-                group_by = config.get_group_by()
-        except Exception as e:
-            print('Can\'t execute group_by function: {}'.format(e))
-        call = FunctionCall(endpoint=endpoint, execution_time=time, version=config.version,
-                            time=datetime.datetime.now(), group_by=str(group_by), ip=ip)
-        db_session.add(call)
-
-
-def get_times():
-    """ Return all entries of measurements with the average and total number of execution times. The results are 
-    grouped by their endpoint. """
-    with session_scope() as db_session:
-        result = db_session.query(FunctionCall.endpoint,
-                                  func.count(FunctionCall.execution_time).label('count'),
-                                  func.avg(FunctionCall.execution_time).label('average')
-                                  ).group_by(FunctionCall.endpoint).order_by(desc('count')).all()
-        db_session.expunge_all()
-        return result
-
-
-def get_hits(date_from=None):
-    """ Return all entries of measurements with the number of execution times which are called after 'date_from'
-    :param date_from: A datetime-object
-    """
-    with session_scope() as db_session:
-        result = db_session.query(FunctionCall.endpoint,
-                                  func.count(FunctionCall.execution_time).label('count')). \
-            filter(FunctionCall.time > date_from).group_by(FunctionCall.endpoint).all()
-        db_session.expunge_all()
-        return result
+        db_session.add(FunctionCall(endpoint=endpoint, execution_time=execution_time, version=config.version, ip=ip))
 
 
-def get_average(date_from=None):
```

Comment:
Why not also keep the `get_average` function? #77 says the plot should include both median and average.